### PR TITLE
rewrite mirror-sdists.sh in python

### LINKED ISSUE #19: json error on mac build
I'm seeing this error.

```
+ ./mirror-sdists.sh:50 	jq --argjson obj '{"type":"build_system","req":"tomli; python_version < \'\''3.11\'\''","why":" -> langchain-0.1.13 -> aiohttp-3.9.3 -> aiosignal-1.3.1 -> frozenlist-1.4.1 -> expandvars-0.12.0 -> hatchling-1.22.3 -> pluggy-1.4.0"}' '. += [$obj]' /Users/dhellmann/Devel/rebuilding-the-wheel/rebuilding-the-wheel/sdists-repo/build-order.json
jq: invalid JSON text passed to --argjson
Use jq --help for help with command-line options,
or see the jq manpage, or online docs  at https://jqlang.github.io/jq
```

Maybe a difference in jq or something else. Now that we build wheels as we go, maybe the build order tracking can be simplified.

### PR DESCRIPTION
**Pull request by dhellmann**
Use multiple python modules in a package for a bit of code organization.

Manage a built-in web server in a thread to avoid having collisions from 
multiple processes.

Use subprocess only when running pip or other tool that has no library 
interface.

Depends-on: #30
Fixes #19

### CODE REVIEW

File: resolve_and_download.py [hunk 1]
```diff
@@ -220,6 +216,12 @@ def main():
 
     VERBOSE = args.verbose
 
+    # Things I want to resolve.
+    requirements = [Requirement(r) for r in args.requirements]
```
Comment (markmc) [inline resolve_and_download.py:222]: Duplicate line

File: mirror_builder/bootstrap.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    # 'pip wheel'.
+    #
+    # https://flit.pypa.io/en/stable/bootstrap.html
+    sdist_filename = sdist.download_sdist(ctx, ['flit_core'])
+    resolved_name = sdist.get_resolved_name(sdist_filename)
+    sdist_root_dir = sdist.unpack_sdist(ctx, sdist_filename)
+    logger.info('building flit_core wheel in %s', sdist_root_dir)
+    subprocess.check_call(
+        ['python3', '-m', 'flit_core.wheel'],
+        cwd=sdist_root_dir,
```
Comment (markmc) [inline mirror_builder/bootstrap.py:23]: What happened to bootstrap_install.py?

**Comment by markmc** [comment]
Looks really great, thanks!

Take a look into the flit bootstrap_install.py thing

File: mirror_builder/bootstrap.py (hunk 2 above)
Comment (dhellmann) [reply mirror_builder/bootstrap.py:23]: That script just unpacks the wheel to install it, but we're already relying on pip to install wheels so we don't need it. It was superfluous in the bash implementation.

**Pull request merged**